code/game/objects/structures/mystery_box.dm

```diff
@@ -0,0 +1,278 @@
+	close_box()
+
+
+/obj/structure/mystery_box/guns
+	desc = "A wooden crate that seems equally magical and mysterious, capable of granting the user all kinds of different pieces of gear. This one seems focused on firearms."
+
+/obj/structure/mystery_box/guns/generate_valid_types()
+	valid_types = GLOB.summoned_guns
+
+/obj/structure/mystery_box/tdome
+	desc = "A wooden crate that seems equally magical and mysterious, capable of granting the user all kinds of different pieces of gear. This one has an extended array of weaponry."
+
+/obj/structure/mystery_box/tdome/generate_valid_types()
+	valid_types = GLOB.mystery_box_guns + GLOB.mystery_box_extended
+
+
+/// This represents the item that comes out of the box and is constantly changing before the box finishes deciding. Can probably be just an /atom or /movable.
+/obj/mystery_box_item
+	name = "???"
+	desc = "Who knows what it'll be??"
+	icon = 'icons/obj/guns/ballistic.dmi'
+	icon_state = "detective"
+	pixel_y = -8
+
+	/// The currently selected item. Constantly changes while choosing, determines what is spawned if the prize is claimed, and its current icon
+	var/selected_path = /obj/item/gun/ballistic/revolver/detective
+	/// The box that spawned this
+	var/obj/structure/mystery_box/parent_box
+	/// Whether this prize is currently claimable
+	var/claimable = FALSE
+
+
+/obj/mystery_box_item/Initialize(mapload)
+	. = ..()
+	var/matrix/starting = matrix()
+	starting.Scale(0.5,0.5)
+	transform = starting
+	add_filter("weapon_rays", 3, list("type" = "rays", "size" = 28, "color" = COLOR_VIVID_YELLOW))
+
+/obj/mystery_box_item/Destroy(force)
+	. = ..()
+	parent_box = null
+
+// this way, clicking on the prize will work the same as clicking on the box
+/obj/mystery_box_item/attack_hand(mob/living/user, list/modifiers)
+	. = ..()
+	if(claimable)
+		parent_box.grant_weapon(user)
+
+/// Start pushing the prize up
+/obj/mystery_box_item/proc/start_animation(atom/parent)
+	parent_box = parent
+
+	var/matrix/starting = matrix()
+	animate(src, pixel_y = 6, transform = starting, time = MBOX_DURATION_CHOOSING, easing = QUAD_EASING | EASE_OUT)
+	loop_icon_changes()
+
+/// Keep changing the icon and selected path
+/obj/mystery_box_item/proc/loop_icon_changes()
+	set waitfor = FALSE
+
+	var/change_delay = 1
+	var/change_delay_delta = 0.2
+	var/start_time = world.time
+
+	/// The uninstantiated item that's currently selected based off selected_path, for use with initial()
+	var/obj/item/selected_item
+
+	while(world.time < start_time + MBOX_DURATION_CHOOSING)
+		change_delay += change_delay_delta
+		selected_path = pick(parent_box.valid_types)
+		selected_item = selected_path
+		icon = initial(selected_item.icon)
+		icon_state = initial(selected_item.icon_state)
+		sleep(change_delay)
+
+	add_filter("ready_outline", 2, list("type" = "outline", "color" = "#FBFF23", "size" = 0.2))
+	name = initial(selected_item.name)
+	parent_box.present_weapon()
+	claimable = TRUE
```

Comment:
I think these could be a series of animates and a timer, would make life a bit easier

Reply:
Wow it just doesn't work. that's depressing

Reply:
Yeah I was messing with it over the last few days with different configurations, but it appears to just take the last one and run it immediately. I'm not thrilled about using all timers instead but meh